qt/aqt/editor.py

```diff
@@ -1091,34 +1091,35 @@ def _processText(self, mime: QMimeData) -> Optional[str]:
             return None
 
         txt = mime.text()
-
-        # inlined data in base64?
-        if txt.startswith("data:image/"):
-            return self.editor.inlinedImageToLink(txt)
-
-        # if the user is pasting an image or sound link, convert it to local
-        if self.editor.isURL(txt):
-            url = txt.split("\r\n")[0]
-            link = self.editor.urlToLink(url)
-            if link:
-                return link
-
-            # not media; add it as a normal link if pasting with shift
-            link = '<a href="{}">{}</a>'.format(url, html.escape(txt))
-            return link
-
-        # normal text; convert it to HTML
-        txt = html.escape(txt)
-        txt = txt.replace("\n", "<br>").replace("\t", " " * 4)
-
-        # if there's more than one consecutive space,
-        # use non-breaking spaces for the second one on
-        def repl(match):
-            return match.group(1).replace(" ", "&nbsp;") + " "
-
-        txt = re.sub(" ( +)", repl, txt)
-
-        return txt
+        processed = ''
```

Comment:
Using an array of strings `processed = [] & "".join(processed)` is exponentially faster than string concatenation: https://waymoot.org/home/python_string/ (Efficient String Concatenation in Python)

Reply:
done. Thanks!